### ISSUE TITLE
Can't assign >64bit quantity to BagHash element

### ISSUE THREAD

**Issue by mscha** [rakudo/rakudo#1983]
## The Problem

A BagHash does appear to support large quantities (> 64 bit).  This works fine:

```
> my $bh = BagHash.new-from-pairs('foo'=>10000000000000000000);
10000000000000000000
```

But if you try to assign a > 64 bit value to an element, it bombs:

```
> $bh<foo>++; say $bh<foo>;
Cannot unbox 64 bit wide bigint into native integer
  in block <unit> at <unknown file> line 1
```

## Environment
* Operating system: Linux
* Compiler version (`perl6 -v`): 

This is Rakudo Star version 2018.04.1 built on MoarVM version 2018.04.1
implementing Perl 6.c.

**Comment by lizmat** [comment]
The underlying issue is actually in the sanity check for assignment:

    my %h is BagHash;
    # dies with Cannot unbox 64 bit wide bigint into native integer
    %h.ASSIGN-KEY("foo",10000000000000000000);

Testing a fix now

**Comment by lizmat** [comment]
Fixed with fd779109dd , tests added with 5afa359cc6 , resolving ticket.

Thanks for reporting!
